docs/DeveloperGuide.md

```diff
@@ -1,34 +1,413 @@
+        `TopicList.showTopics()` method.
+Step 2: The `TopicList.showTopics()` method then calls the method `Ui.printTopicList()`. The 
+        current `TopicList` is passed into the called method.
+Step 3: The `Ui.printTopicList()` method then prints out the description of each topic in the 
+        `TopicList`. 
+
+`NoteList` is also an `ArrayList` of type `Displayable`, which is one of two interfaces implemented in the code for 
+E-Duke-8. As such, many of the commands that manipulate the `TopicList` make use of the package `java.util.ArrayList`. 
+The `NoteList` stores `Note` objects. Each topic has 1 `NoteList`. 
+
+#### 2.2.3. Implementation of Notes
+
+**Adding a new note:**
+
+This task is performed by the `NoteList.add()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then calls the `NoteList.add()` method. 
+        A new `Note` object is passed into its parameter.
+
+Step 2: The `NoteList.add()` method makes use of `ArrayList` API, specifically the `ArrayList.add()` method, to add 
+        the `Note` object into `NoteList`.
+
+**Deleting a note:**
+
+This task is performed by the `NoteList.delete()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then calls the `NoteList.delete()` method. 
+        An integer that represents the index of the `Note` object to be deleted within the `NoteList` is passed into 
+        this method.
+
+Step 2: The `NoteList.delete()` method makes use of `ArrayList` API, specifically the `ArrayList.remove()` method, to 
+        delete the `Note` object in `NoteList`.
+
+**Listing out all notes in a topic**
+
+This task is performed by the `Ui.printNoteList()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then instantiates an Ui object
+and calls the `Ui.listInteraction` method. 
+
+Step 2: The `Ui.listInteraction` method calls the `Ui.printNoteList()` method. The topic's `NoteList` is passed into 
+this method. `Ui.printNoteList()` prints out the descriptions and texts of all the `Note` objects in the 
+topic's `NoteList`.
+
+#### 2.2.4. Design of Option and OptionList 
+
+The `Option` and `OptionList` classes implements the `Displayable` and `DisplayableList` interfaces respectively. 
+The `Option` object stores one option of a question while the `OptionList` object stores all 4 options of the same 
+question. The class diagram below illustrates the structure of both classes. 
+
+![Option_and_OptionList_Class](./images/Option.png)
+
+### 2.3. Logic Component
+
+The main application logic, such as provisioning quizes, is handled by the Logic component. This component also acts as the middleman between the backend and frontend by processing data before passing it to the user interface and parsing user input from the user interface.
+
+#### 2.3.1. Design of Parser
+
+![Parser Diagram](./images/ParserDiagram.png)
```

Comment:
The diagram is very well made, but it is rather tough to view on the webpage. Perhaps you could split the MenuParser commands and QuizParser commands into two diagrams? The various commands could also be arranged around the abstract Command class as a semi circle or a square with no bottom so that the width of the picture can be greatly reduced, so maybe that will help with the issue.